src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2Serializer.cs

```diff
@@ -2250,6 +2518,89 @@ protected virtual void WriteSubjectLocality(XmlWriter writer, Saml2SubjectLocali
             writer.WriteEndElement();
         }
 
+        private EncryptedData CreateEncryptedDataHelper(SymmetricSecurityKey key, string algorithm, byte[] assertionData, EncryptingCredentials encryptingCredentials)
```

Comment:
>private E [](start = 8, length = 9)

it seems to me like the functions for creating the helper which essentially returns Encrypted data (I'm a bit confused about that), should be in another class?

Reply:
I'll think about refactoring these.

Reply:
addressed with: ac218685781cbc736b39914f3afec69f16063937